annotated_test.go

```diff
@@ -983,4 +984,519 @@ func TestAnnotate(t *testing.T) {
 		assert.Contains(t, err.Error(), "invalid annotation function func(fx_test.B) string")
 		assert.Contains(t, err.Error(), "fx.In structs cannot be annotated")
 	})
+
+	t.Run("Hooks", testHookAnnotations)
+}
+
+func testHookAnnotations(t *testing.T) {
+	t.Parallel()
+
+	validateApp := func(t *testing.T, opts ...fx.Option) error {
+		return fx.ValidateApp(
+			append(opts, fx.Logger(fxtest.NewTestPrinter(t)))...,
+		)
+	}
+
+	t.Run("depend on result interface of target", func(t *testing.T) {
+		//t.Skip()
+		type stub interface {
+			String() string
+		}
+
+		var started bool
+
+		app := fxtest.New(t,
+			fx.Provide(
+				fx.Annotate(
+					func() (stub, error) {
+						b := []byte("expected")
+						return bytes.NewBuffer(b), nil
+					},
+					fx.OnStart(func(_ context.Context, s stub) error {
+						started = true
+						if !assert.Equal(t, "expected", s.String()) {
+							return fmt.Errorf(
+								"expected %q got %q",
+								"expected",
+								s.String(),
+							)
+						}
```

Comment:
This will fail the test directly instead of waiting for `require.NoError`

```suggestion
						require.Equal(t, "expected", s.String()) 
```